interfaces/prompting/requestrules/requestrules.go

```diff
@@ -88,15 +88,17 @@ func (rule *Rule) Expired(currentTime time.Time) bool {
 	return false
 }
 
-// variantEntry stores the variant struct and the ID of its corresponding rule.
+// variantEntry stores the actual pattern variant struct which can be used to
+// match paths, and the set of rule IDs whose path patterns render to this
+// variant. All rules in a particular entry must have the same outcome, and
+// that outcome is stored directly in the variant entry itself.
 //
-// This is necessary since multiple variants might render to the same string,
-// and it would be necessary to make a deep comparison of two variants to tell
-// that they are the same. Since we want to map from variant to rule ID, we need
-// to use the variant string as the key.
+// Use the rendered string as the key for this entry, since pattern variants
+// cannot otherwise be easily checked for equality.
 type variantEntry struct {
 	Variant patterns.PatternVariant
-	RuleID  prompting.IDType
+	Outcome prompting.OutcomeType
+	RuleIDs map[prompting.IDType]bool
```

Comment:
this is not stored to disk, right?

Reply:
Correct. The rule tree uses `variantEntry` as the leaves, and this whole tree is in-memory only, used for efficient lookup and precedence decision (since precedence between two rules depends on the particular pattern variant which matched a requested path).